hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FSSchedulerNode.java

```diff
@@ -110,6 +132,44 @@ synchronized FSAppAttempt getReservedAppSchedulable() {
   }
 
   /**
+   * List reserved resources after preemption and assign them to the
+   * appropriate applications in a FIFO order.
+   * @return if any resources were allocated
+   */
+  synchronized boolean assignContainersToPreemptionReservees() {
```

Comment:
The reason I brought it here is that we need to run it synchronized to make sure the list does not change. The preemption reservation data belongs here I think. What I could do is that do the assignContainersToPreemptionReservees tasks in fair scheduler and expose the lock to that function.